crates/sui-indexer-alt-graphql/src/api/types/object.rs

```diff
@@ -174,20 +180,23 @@ impl Object {
         if bounds > 1 {
             Err(bad_user_input(Error::OneBound))
         } else if let Some(v) = key.version {
-            Ok(Self::at_version(ctx, key.address, v).await?)
+            Ok(Self::at_version(ctx, scope, key.address, v).await?)
         } else if let Some(v) = key.root_version {
-            Ok(Self::version_bounded(ctx, key.address, v).await?)
+            Ok(Self::version_bounded(ctx, scope, key.address, v).await?)
         } else if let Some(cp) = key.at_checkpoint {
-            Ok(Self::checkpoint_bounded(ctx, key.address, cp).await?)
+            let scope = scope.with_checkpoint_viewed_at(cp.into());
+            Ok(Self::checkpoint_bounded(ctx, scope, key.address, cp).await?)
```

Comment:
Technically, we no longer need `atCheckpoint`, because now, these two queries are equivalent:

```graphql
query A($A: SuiAddress!, $C: UInt53) {
  object(address: $A, atCheckpoint: $C) { ... }
}

query A($A: SuiAddress, $C: UInt53) {
  checkpoint(sequenceNumber: $C) { query { object(address: $A) { ... } } }
}
```

but opting to keep `atCheckpoint` as an option because it's less verbose and nested, even though it could cause some confusion when combined with `checkpoint ... query` (because the `atCheckpoint` parameter will take precedence over the scope's `checkpoint_viewed_at`).

Reply:
Flattening `Query` into `Checkpoint` would make this better, but it would make it quite difficult to figure out which fields a `Checkpoint` has -- I don't think there's a major reason to use `atCheckpoint` instead of nesting within `Checkpoint.query` other than convenience (but I also don't think there's a great need to make sure there's only one way to perform each action).